java/ingest/ingest-core/src/main/java/sleeper/ingest/job/status/IngestJobStatusType.java

```diff
@@ -17,31 +17,26 @@
 package sleeper.ingest.job.status;
 
 import sleeper.core.record.process.status.ProcessFailedStatus;
-import sleeper.core.record.process.status.ProcessFinishedStatus;
+import sleeper.core.record.process.status.ProcessRun;
 import sleeper.core.record.process.status.ProcessStatusUpdate;
 
-import java.util.stream.Stream;
-
 /**
  * Defines the states an ingest job can be in.
  */
 public enum IngestJobStatusType {
-    REJECTED(IngestJobRejectedStatus.class, 1),
-    ACCEPTED(IngestJobAcceptedStatus.class, 2),
-    FAILED(ProcessFailedStatus.class, 3),
-    IN_PROGRESS(IngestJobStartedStatus.class, 4),
-    FINISHED(ProcessFinishedStatus.class, 5);
-
-    private final Class<?> statusUpdateClass;
-    private final int order;
-
-    IngestJobStatusType(Class<?> statusUpdateClass, int order) {
-        this.statusUpdateClass = statusUpdateClass;
-        this.order = order;
-    }
+    REJECTED(1, 5),
+    ACCEPTED(2, 1),
+    FAILED(3, 6),
+    IN_PROGRESS(4, 2),
+    UNCOMMITTED(5, 3),
+    FINISHED(6, 4);
 
-    public int getOrder() {
-        return order;
+    private final int orderInJob;
+    private final int orderInRun;
+
+    IngestJobStatusType(int orderInJob, int orderInRun) {
```

Comment:
Can we have more documentation? It's not clear what the difference is between a job and a run?

Reply:
Good points. When we're talking about a job, if one of the runs of that job failed it doesn't fail the whole job. When we're talking about a run, then if there's a failure anywhere in the run the whole run has failed. There's some explanation below on StatusTracker, but it should be much clearer.

Reply:
I've split it into two classes, IngestJobStatusType and IngestJobUpdateType. Hopefully it's clearer now.